CardanoSharp Unity Project/Assets/Scripts/CardanoManager.cs

```diff
@@ -38,39 +46,94 @@ void Start()
 
     }
 
+    #region Wallet Creation
     public Mnemonic GenerateMnemonic(int size = 24, WordLists wordLists = WordLists.English)
     {
         return _mnemonicService.Generate(size, wordLists);
     }
 
+    public Mnemonic RestoreMnemonic(string words)
+    {
+        return _mnemonicService.Restore(words);
+    }
+
+    public IAccountNodeDerivation GetAccountNode(string words)
+    {
+        Mnemonic mnemonic = _mnemonicService.Restore(words);
+
+        // Fluent derivation API
+        return mnemonic
+            .GetMasterNode()                // IMasterNodeDerivation
+            .Derive(PurposeType.Shelley)    // IPurposeNodeDerivation
+            .Derive(CoinType.Ada)           // ICoinNodeDerivation
+            .Derive(0);                     // IAccountNodeDerivation
+    }
+
+    public KeyPair GenerateKeyPair()
+    {
+        return KeyPair.GenerateKeyPair();
+    }
+    #endregion
+
+    #region Encryption
+    public string EncryptKeys(PrivateKey accountPrivateKey, PublicKey accountPublicKey)
+    {
+        return JsonSerializer.Serialize((accountPrivateKey.Encrypt("spending_password"), accountPublicKey));
+    }
+
+    public (PrivateKey, PublicKey) DecryptKeys(string accountNode)
+    {
+        var load = JsonSerializer.Deserialize<(PrivateKey, PublicKey)>(accountNode);
+        return (load.Item1.Decrypt("spending_password"), load.Item2);
+    }
```

Comment:
@tweakch im thinking we actually store it here too. I think this "manager" can handle the persistence of the items we deal with. thoughts?